design/deploy.md

```diff
@@ -0,0 +1,98 @@
+# Deploy
+
+A proposal to add support for hosting service deployment to the Polymer CLI.
+
+## Goals
+- Provide new users with a fast way to get a project hosted online.
+- Provide advanced users with an initial bootstrap for more advanced deployments.
+- Demonstrate the [PRPL](https://www.polymer-project.org/2.0/toolbox/server) pattern where HTTP/2 Server Push is available.
+
+## Non-goals
+- Do not try to support every possible deployment environment and configuration, or even a lot of them. Just enough to get users started.
+
+## Design
+
+- Add an `--environment` flag.
+  - Every command knows that it's being run in the context of this environment.
+  - Under the `default` environment, all commands operate as they do today.
+  - Under other environments:
+     - `build` now outputs the boilerplate needed to deploy the project to a given service, in addition to the traditional app build output.
+     - `serve` runs an appropriate HTTP server that emulates the configured hosting environment, instead of polyserve.
```

Comment:
I hope `polymer serve` isn't too intrinsically married to trafficking parameters over to `polyserve` to get its juice from an environment specification.  I really like the idea that `serve` could utilize different backends in this way like running a local GAE emulator or a local docker instance.